### PR TITLE
Remove intermediate docker files upon exit

### PR DESCRIPTION
**Pull request by jxyang**
Signed-off-by: Xuejun Yang <xuejya@microsoft.com>

### CODE REVIEW

File: .azure_pipelines/ci-pipeline-makefile.yml [hunk 1]
```diff
@@ -73,6 +73,7 @@ jobs:
           make clean -C $(Build.SourcesDirectory)/tests
           make clean -C $(Build.SourcesDirectory)/solutions
           sudo rm -rf $(Build.SourcesDirectory)/build/tests
+          docker system prune -f
```
Comment (vtikoo) [inline .azure_pipelines/ci-pipeline-makefile.yml:76]: Why only intermediate images? Can we additionally use `-a` option here, that would delete all unused images.

Comment (jxyang) [reply .azure_pipelines/ci-pipeline-makefile.yml:76]: That probably will increase the average pipeline runtime. I am not quite sure how the docker images are cached and reused for the next image building action.

**Comment by jxyang** [comment]
This PR meant to fix nightly pipeline failures. `Disk out of space` is just one of them. More investigations are needed.

Converting to draft for now.

**Comment by jxyang** [comment]
The root cause of the problem is `appbuilder` doesn't delete the exported archives of docker images. This can fill up the `/tmp` directory pretty quickly.

Keep the `docker system prune -f` as an extra measure to save disk space. Also use `df` to report free disk space at the beginning of pipeline run.

**Review by vtikoo — APPROVED** [review]

**Pull request merged**